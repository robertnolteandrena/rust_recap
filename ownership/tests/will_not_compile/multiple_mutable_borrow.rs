fn main() {
    let mut a = String::from("Hello");
    let mutable_borrow_1 = &mut a;
    let mutable_borrow_2 = &mut a;
    println!("{}, {}", mutable_borrow_1, mutable_borrow_2);
}
