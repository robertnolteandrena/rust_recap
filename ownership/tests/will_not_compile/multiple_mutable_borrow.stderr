error[E0499]: cannot borrow `a` as mutable more than once at a time
 --> tests/will_not_compile/multiple_mutable_borrow.rs:4:28
  |
3 |     let mutable_borrow_1 = &mut a;
  |                            ------ first mutable borrow occurs here
4 |     let mutable_borrow_2 = &mut a;
  |                            ^^^^^^ second mutable borrow occurs here
5 |     println!("{}, {}", mutable_borrow_1, mutable_borrow_2);
  |                        ---------------- first borrow later used here
